### PR TITLE
Add vmstat metrics to meminfo_numa

### PR DESCRIPTION
**Pull request by kwisniewski98**
Signed-off-by: Wisniewski, Krzysztof2 <Krzysztof2.Wisniewski@intel.com>

### CODE REVIEW

File: collector/meminfo_numa_linux.go [hunk 1]
```diff
@@ -116,6 +122,12 @@ func getMemInfoNuma() ([]meminfoMetric, error) {
 			return nil, err
 		}
 		metrics = append(metrics, numaStat...)
+
+		vmStat, err := parseVMStatNuma(vmstatFile, nodeNumber[1])
```
Comment (katarzyna-z) [inline collector/meminfo_numa_linux.go:127]: Are you sure that at this point `nodeNumber` contains at least 2 elements?

**Review by katarzyna-z — COMMENTED** [review]
It will be good to add unit tests for the code from this pull request.

File: collector/meminfo_numa_linux.go [hunk 2]
```diff
@@ -106,8 +106,14 @@ func getMemInfoNuma() ([]meminfoMetric, error) {
...
 
+		vmstatFile, err := os.Open(filepath.Join(node, "vmstat"))
+		if err != nil {
+			return nil, err
+		}
+		defer vmstatFile.Close()
+
 		nodeNumber := meminfoNodeRE.FindStringSubmatch(node)
-		if nodeNumber == nil {
+		if nodeNumber == nil || len(nodeNumber) != 2 {
```
Comment (Creatone) [inline collector/meminfo_numa_linux.go]: Maybe you can put that number into constant? Something like `expectedNodeNumberLength`.

File: collector/meminfo_numa_linux.go [hunk 3]
```diff
@@ -181,3 +193,29 @@ func parseMemInfoNumaStat(r io.Reader, nodeNumber string) ([]meminfoMetric, erro
...
+		if len(parts) != 2 {
+			return nil, fmt.Errorf("line scan did not return 2 fields: %s", line)
+		}
+
+		fv, err := strconv.ParseFloat(parts[1], 64)
+		if err != nil {
+			return nil, fmt.Errorf("invalid value in vmstat: %w", err)
+		}
+
+		vmstat = append(vmstat, meminfoMetric{"vmstat_" + parts[0], prometheus.UntypedValue, nodeNumber, fv})
```
Comment (Creatone) [inline collector/meminfo_numa_linux.go]: Why you use `prometheus.UntypedValue`?

```
nr_free_pages 7637089                                                                                                                                                                                                                                         
nr_zone_inactive_anon 66462                                                                                                                                                                                                                                   
nr_zone_active_anon 1081045                                                                                                                                                                                                                                   
nr_zone_inactive_file 680625
nr_zone_active_file 424565
nr_zone_unevictable 163288
nr_zone_write_pending 2661
nr_mlock 469
nr_page_table_pages 11221
nr_kernel_stack 15616
nr_bounce 0
nr_zspages 0
nr_free_cma 0
numa_hit 24162905
numa_miss 0
numa_foreign 0
numa_interleave 22993
numa_local 24162905
numa_other 0
nr_inactive_anon 66462
nr_active_anon 1081045
nr_inactive_file 680625
nr_active_file 424565
nr_unevictable 163288
nr_slab_reclaimable 83821
nr_slab_unreclaimable 46887
nr_isolated_anon 0
nr_isolated_file 0
workingset_nodes 0
workingset_refault 0
workingset_activate 0
workingset_restore 0
workingset_nodereclaim 0
nr_anon_pages 1079658
nr_mapped 211563
nr_file_pages 1336365
nr_dirty 2661
nr_writeback 0
nr_writeback_temp 0
nr_shmem 230785
nr_shmem_hugepages 0
nr_shmem_pmdmapped 0
nr_file_hugepages 0
nr_file_pmdmapped 0
nr_anon_transparent_hugepages 0
nr_unstable 0
nr_vmscan_write 0
nr_vmscan_immediate_reclaim 0
nr_dirtied 622644
nr_written 612737
nr_kernel_misc_reclaimable 0
nr_foll_pin_acquired 0
nr_foll_pin_released 0
```
In my opinion, these values are gauges.

**Review by Creatone — CHANGES_REQUESTED** [review]
Check this out

**Pull request closed without merging**